### PR TITLE
small moth changes

### PR DESCRIPTION
**Pull request by LankLTE**
moths can eat prisoner jumpsuits, berets and head hats. ghost burger is edible since it uses cloth
can no longer eat magboots, face items (like gas masks), or anything with items stored in it 
they can now squeak

image

**Changelog**

:cl: Lank
- tweak: What Moths consider to be made of cloth should now be more accurate. 
- tweak: Moths are now able to squeak.

### CODE REVIEW

**Comment by OctoRocket** [comment]
Why specifically can't they eat winter coats?

**Comment by LankLTE** [comment]
> Why specifically can't they eat winter coats?

they have storage, which allows you to eat any item that can be put in them (like hypos, door remotes, nuke disk, etc.)

**Comment by CrigCrag** [comment]
I'm not sure if they were also changed, but if storage is the no-go for edible items, make sure web vests, jensen/gentle coats, bomber jacket and so on aren't edible.

**Comment by deltanedas** [comment]
combat boots applies for knife
this should probably just be a check on food when eating it so anything that has items cant be eaten

**Comment by LankLTE** [comment]
> I'm not sure if they were also changed, but if storage is the no-go for edible items, make sure web vests, jensen/gentle coats, bomber jacket and so on aren't edible.

outerwear was already restricted to winter coats so this is fine

> combat boots applies for knife
this should probably just be a check on food when eating it so anything that has items cant be eaten

Not sure how easy that would be to implement nor do I really think it's worth it, but either way I think the combat boots are fine since it can only eat knives which is far less abusable

**Comment by mirrorcult** [comment]
shouldn't be trying to catch this on a case-by-case basis, either eating stuff with containers should be disallowed or the container contents should be dropped

**Comment by LankLTE** [comment]
> shouldn't be trying to catch this on a case-by-case basis, either eating stuff with containers should be disallowed or the container contents should be dropped

I can work on that later, but at the moment it’s probably good to stop people from deleting items still

**Comment by deltanedas** [comment]
check medibot construction emptying i think that would apply here

**Comment by LankLTE** [comment]
> shouldn't be trying to catch this on a case-by-case basis, either eating stuff with containers should be disallowed or the container contents should be dropped

eating containers is now disallowed

File: Content.Server/Nutrition/EntitySystems/FoodSystem.cs [hunk 1]
```diff
@@ -119,6 +120,12 @@ private void OnFeedFood(EntityUid uid, FoodComponent foodComponent, AfterInterac
                 return (false, true);
             }
 
+            // Check for storage on the food item
+            if (EntityManager.TryGetComponent<ServerStorageComponent>(food, out var storageState))
```
Comment (deltanedas) [inline Content.Server/Nutrition/EntitySystems/FoodSystem.cs:124]: TryComp

Comment (LankLTE) [reply Content.Server/Nutrition/EntitySystems/FoodSystem.cs]: I now use TryComp since I'm actually using storageState

**Comment by LankLTE** [comment]
they can now eat winter coats again, just not if anything is stored in them

**Comment by keronshb** [comment]
> > Why specifically can't they eat winter coats?
> 
> they have storage, which allows you to eat any item that can be put in them (like hypos, door remotes, nuke disk, etc.)

You can allow them to eat winter coats. Just when the coat is deleted, force the items to drop.

**Comment by LankLTE** [comment]
> You can allow them to eat winter coats. Just when the coat is deleted, force the items to drop.

I just changed it to disallow them from eating coats that have anything stored inside. I think that works just as well.

**Comment by dmnct** [comment]
> > You can allow them to eat winter coats. Just when the coat is deleted, force the items to drop.
> 
> I just changed it to disallow them from eating coats that have anything stored inside. I think that works just as well.

Are players going to realize why they could not eat the coat? This is not intuitive. Dropping the item seems like a better solution

**Comment by Emisse** [comment]
its fine it says that it cant be eat because an item is stored inside

**Pull request merged**